src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+  let temporaryPreviousObject = { ...state };
+
+  for (const item of actions) {
+    const type = item.type;
+
+    switch (type) {
+      case 'addProperties':
```

Comment:
Not critical but would be great to use variables instead of the hardcoded values